pettingzoo/butterfly/prospector/prospector.py

```diff
@@ -764,7 +764,7 @@ def state(self):
         return state
 
     def step(self, action):
-        if self.dones[self.agent_selection]:
+        if self.terminations[self.agent_selection]:
```

Comment:
This needs to be `if self.terminations[self.agent_selection] or self.truncations[self.agent_selection]` I think.